src/openslide-vendor-philips-isyntax.c

```diff
@@ -0,0 +1,397 @@
+        // But we don't need much. Axis-aligned lines, and some simple text.
+        int pad = 1;
+        uint32_t color = 0xff0000ff; // ARGB
+        draw_horiz_line(tile_pixels, tile_width, /*y=*/pad, /*start=*/pad, /*end=*/tile_width - pad, color);
+        draw_horiz_line(tile_pixels, tile_width, /*y=*/tile_height - pad, /*start=*/pad, /*end=*/tile_width - pad, color);
+
+        draw_vert_line(tile_pixels, tile_width, /*x=*/pad, /*start=*/pad, /*end=*/tile_height - pad, color);
+        draw_vert_line(tile_pixels, tile_width, /*x=*/tile_width - pad, /*start=*/pad, /*end=*/tile_height - pad, color);
+
+        char buf[128];
+        sprintf(buf, "x=%d,y=%d,s=%d", tile_row, tile_col, scale);
+        draw_text(tile_pixels, tile_width, 10, 10, color, buf);
+    }
+}
+
+static bool philips_isyntax_detect(
+        const char *filename,
+        struct _openslide_tifflike *tl G_GNUC_UNUSED,
+        GError **err G_GNUC_UNUSED) {
+    LOG("got filename %s", filename);
+    LOG_VAR("%p", tl);
+    // reject TIFFs
+    if (tl) {
+        g_set_error(err, OPENSLIDE_ERROR, OPENSLIDE_ERROR_FAILED,
+                    "Is a TIFF file");
+        return false;
+    }
+
+    g_autoptr(_openslide_file) f = _openslide_fopen(filename, err);
+    if (f == NULL) {
+        LOG("Failed to open file");
+        return false;
+    }
+
+    const int num_chars_to_read = 256;
+    g_autofree char *buf = g_malloc(num_chars_to_read);
+    size_t num_read = _openslide_fread(f, buf, num_chars_to_read-1);
+    buf[num_chars_to_read-1] = 0;
+    LOG_VAR("%ld", num_read);
+    LOG_VAR("%s", buf);
+
+    // TODO(avirodov): probably a more robust XML parsing is needed.
+    if (strstr(buf, "<DataObject ObjectType=\"DPUfsImport\">") != NULL) {
+        LOG("got isyntax.");
+        return true;
+    }
+
+    LOG("not isyntax.");
+    return false;
+}
+
+static bool philips_isyntax_read_tile(
+        openslide_t *osr,
+        cairo_t *cr,
+        struct _openslide_level *osr_level,
+        int64_t tile_col, int64_t tile_row,
+        void *arg G_GNUC_UNUSED,
+        GError **err G_GNUC_UNUSED) {
+    philips_isyntax_t *data = osr->data;
+    isyntax_t* isyntax = data->isyntax;
+
+    philips_isyntax_level* pi_level = (philips_isyntax_level*)osr_level;
+
+    // LOG("level=%d tile_col=%ld tile_row=%ld", pi_level->level_idx, tile_col, tile_row);
+    // tile size
+    int64_t tw = libisyntax_get_tile_width(isyntax);
+    int64_t th = libisyntax_get_tile_height(isyntax);
+
+    // Openslide cache
+    g_autoptr(_openslide_cache_entry) cache_entry = NULL;
+    uint32_t *tiledata = _openslide_cache_get(osr->cache,
+                                              pi_level, tile_col, tile_row,
+                                              &cache_entry);
+    if (!tiledata) {
+        int scale = libisyntax_level_get_scale(pi_level->isyntax_level);
+        ASSERT_OK(libisyntax_tile_read(isyntax, data->cache->cache, scale, tile_col, tile_row, &tiledata));
```

Comment:
With the current libisyntax implementation, we need to byteswap the result on big-endian systems.  (Neither libisyntax nor OpenSlide are currently being tested on such systems, though, so it's possible that there are more fundamental issues as well.)

Reply:
Is there anything actionable I can do here? Just checking. Please unresolve if there is.

Maybe it will be somewhat addressed by having an `output_format=RGBA or BGRA` flag that we are discussing in libisyntax thread. That flag could include endianness later on (e.g. ABGR if needed).

Reply:
Eventually big-endian systems should be handled either in libisyntax or here.  libisyntax is probably the correct place for it, so hopefully OpenSlide won't have to do anything.  I'll unresolve until we have a solid answer here.

I've acquired access to a s390x machine for big-endian testing.  I can help run tests when this work gets further along.